test/e2e/storage/persistent_volumes.go

```diff
@@ -313,6 +320,55 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
 				pvc = nil
 			})
 		})
+
+		ginkgo.Context("delete pv prior", func() {
+			ginkgo.AfterEach(func() {
+				framework.Logf("AfterEach: Cleaning up test resources.")
+				if errs := e2epv.PVPVCCleanup(c, ns, pv, pvc); len(errs) > 0 {
+					framework.Failf("AfterEach: Failed to delete PVC and/or PV. Errors: %v", utilerrors.NewAggregate(errs))
+				}
+			})
+
+			// Waiting for the pvc and pv bound and then delete the pv before pvc
+			ginkgo.It("to exercise deletion of PV prior to deletion of PVC for a Bound PV-PVC pair", func() {
+
+				ctx, cancel := context.WithTimeout(context.Background(), csiPodRunningTimeout)
+				defer cancel()
+
+				pv, pvc, err = e2epv.CreatePVPVC(c, f.Timeouts, pvConfig, pvcConfig, ns, false)
+				framework.ExpectNoError(err)
+				// 1. verify that the PV and PVC are bound
+				ginkgo.By("Validating the PV-PVC binding")
+				framework.ExpectNoError(e2epv.WaitOnPVandPVC(c, f.Timeouts, ns, pv, pvc))
+
+				if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.HonorPVReclaimPolicy) {
+					// 2. wait for finalizer to be added to pv
+					ginkgo.By(fmt.Sprintf("Wait for finalizer %s to be added to pv %s", pvDeletionInTreeProtectionFinalizer, pv.Name))
+					err = e2epv.WaitForPVFinalizer(ctx, c, pv.Name, pvDeletionInTreeProtectionFinalizer, 1*time.Millisecond, 1*time.Minute)
+					framework.ExpectNoError(err)
+				}
+
+				// 3. delete pv
+				ginkgo.By("delete pv")
+				err = e2epv.DeletePersistentVolume(c, pv.Name)
+				framework.ExpectNoError(err)
+
+				// 4. delete pvc
+				ginkgo.By("delete pvc")
+				err = e2epv.DeletePersistentVolumeClaim(c, pvc.Name, ns)
+				framework.ExpectNoError(err)
+
+				// 5. waiting for pv to be deleted
+				ginkgo.By("Wating for the pv to be deleted")
```

Comment:
This looks odd. There is te finalizer and at the same time there is no in-tree nor external provisioner that would delete the volume in the storage backend + remove the finalizer. I would expect that the PV exists and is never removed - that's goal of the feature, right?

Then I would remove the finalizer + re-check the PV is deleted.

Reply:
+1
Just to add a bit more clarity to Jan's comment, he mentioned that there is no deletion support of in-tree volumes of iscsi and nfs types. In theory, the expectation should be that the PV remains in the inventory as the finalizer prevents it from getting deleted.

Reply:
so, If we add the finalizer  and  delete the pv, it would expect that the pv exists exists, like this? CC  @xing-yang

Reply:
@jsafrane so the tes is

1. create pv and pvc, wait pv-pvc bound
2. wait finalizer to be added to pv
3. delete pv 
4. check the pv whether it's exists
5. remove the finalizer
6. re-check the pv

Reply:
After (4) delete PVC as well, since nfs in-tree volumes do not have a deleter plugin, the PV should still be present.

Reply:
@deepakkinni Should we need to wait the pvc to be deleted?

Reply:
yup.